include/libtorrent/aux_/win_crypto_provider.hpp

```diff
@@ -123,40 +121,37 @@ namespace libtorrent { namespace aux
 			TORRENT_ASSERT(size == DWORD(digest_size));
 		}
 	private:
-		void create()
+		HCRYPTHASH create()
 		{
-			TORRENT_ASSERT(!m_initialized);
-
-			if (CryptCreateHash(get_provider(), AlgId, 0, 0, &m_hash) == false)
+			HCRYPTHASH ret;
+			if (CryptCreateHash(get_provider(), AlgId, 0, 0, &ret) == false)
 			{
 #ifndef BOOST_NO_EXCEPTIONS
 				throw system_error(error_code(GetLastError(), system_category()));
 #else
 				std::terminate();
 #endif
 			}
-			m_initialized = true;
+			return ret;
 		}
 
 		void destroy()
 		{
-			if (m_initialized) CryptDestroyHash(m_hash);
-			m_initialized = false;
+			CryptDestroyHash(m_hash);
```

Comment:
how about replacing all calls to ``destroy()`` with calls to ``CryptDestroyHash(m_hash);``? (and removing this function)